### PR TITLE
ENH: Add mnelab, polars, update to vtk 9.3 and PySide6

### LINKED ISSUE #82: Add MNELAB (requires Qt 6)
As soon as we have Qt6 on `conda-forge`

To keep track: https://github.com/conda-forge/qt-main-feedstock/pull/14

### LINKED ISSUE #250: Include Polars
Has started to replace Pandas in new projects:

https://pola.rs

### PR DESCRIPTION
**Pull request by larsoner**
Closes #82
Closes #250

### CODE REVIEW

**Comment by hoechenberger** [comment]
@larsoner The problem here seems to that `graphviz` pins to `libexpat >=2.6.2`, but `vtk-base` needs an older version (`<2.6`)

**Comment by hoechenberger** [comment]
Actually … no, disregard my previous comment, this doesn't check out at least on my Apple Silicon Mac:

```shell
❯ conda search vtk-base==9.3.0 --info |grep libexpat
  - libexpat >=2.6.2,<3.0a0
  - libexpat >=2.6.2,<3.0a0
  - libexpat >=2.6.2,<3.0a0
  - libexpat >=2.6.2,<3.0a0
  - libexpat >=2.6.2,<3.0a0
❯ conda search graphviz==11.0.0 --info |grep libexpat
  - libexpat >=2.6.2,<3.0a0
```

**Comment by larsoner** [comment]
Now it looks like at least https://github.com/conda-forge/liblsl-feedstock/pull/11 is in the way

**Comment by larsoner** [comment]
Now held up by https://github.com/conda-forge/libtheora-feedstock/issues/21, https://github.com/conda-forge/pyobjc-framework-fsevents-feedstock/pull/39, https://github.com/conda-forge/pyobjc-framework-coreservices-feedstock/pull/20, etc.

**Comment by larsoner** [comment]
macos-11 is no longer available, I'll tweak then we should be good @hoechenberger , feel free to mark for merge-when-green if you're happy!

I dropped yasa because it's blocking a SciPy upgrade and they haven't merged my PR to fix things (and then cut a release), hope that's okay

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,10 +34,6 @@ jobs:
       - uses: actions/checkout@v3
       - uses: mamba-org/setup-micromamba@v1
         with:
-          condarc: |
-            channels:
-              - conda-forge
-            channel_priority: strict
```
Comment (larsoner) [inline .github/workflows/build.yml:40]: Pretty sure we don't need these based on them saying:

> If nothing is specified, setup-micromamba will create a .condarc next to the micromamba binary with conda-forge as the only channel.

So it should only use conda-forge I think?

Comment (hoechenberger) [reply .github/workflows/build.yml:40]: Sounds reasonable to me!

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -72,19 +68,19 @@ jobs:
...
 
   build_linux:
-    name: Build (Linux)
-    runs-on: ubuntu-20.04
+    name: Build (${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04]  # Intel and Apple Silicon, respectively
```
Comment (hoechenberger) [inline .github/workflows/build.yml:76]: this comment seems wrong?

**Comment by larsoner** [comment]
Ooof looks like there is a problem with vtk 9.3.0 (and upward) -- it's built against `qt6-main`. One way forward would be to follow https://github.com/conda/constructor/issues/515 and package our own PyQt6. WDYT @hoechenberger ?

**Comment by larsoner** [comment]
... or just hope people don't use plain Python + interactive mode + plotting and stick with PySide6 :(

**Comment by hoechenberger** [comment]
@larsoner More and more conda-forge packages are migrating to qt6 (and hence pyside6). So this is something we'll have to deal with anyway. An alternative would be to switch our installers away from conda and install everything from PyPI. This should work for all platforms except for aarch64. It's what I do in my dev container PR except for vtk and a few other packages (since they're missing for aarch64 on PyPI).

**Comment by larsoner** [comment]
Or maybe someday https://github.com/conda-forge/pyqt-feedstock/pull/136 will land :crossed_fingers:

**Comment by hoechenberger** [comment]
I find this an extremely annoying situation esp since PySide6 is the official Qt binding. Why isn't plotting working properly with it???!

cc @cbrnr

**Comment by larsoner** [comment]
Same as before, https://bugreports.qt.io/browse/PYSIDE-2192

**Comment by hoechenberger** [comment]
yep...

**Comment by hoechenberger** [comment]
@cbrnr Have you had any chance to test whether it's working on Py3.13?

**Comment by cbrnr** [comment]
Not yet, but I will try!

**Comment by cbrnr** [comment]
Unfortunately, PySide6 does not support 3.13 yet.

**Comment by larsoner** [comment]
This is close. One blocker is actually Spyder because it explicitly depends on `pyqt`.

**Comment by cbrnr** [comment]
But why is pyqt problematic?

**Comment by larsoner** [comment]
Because there are only PyQt5 builds (no PyQt6) which means it pulls in qt-main *in addition to* qt6-main and this causes problems

**Comment by cbrnr** [comment]
Ah, I misread that as qtpy. Too many qt py side 5 6 I guess 😄. It's weird that Spyder does not only depend on qtpy though, since I thought they develop and maintain qtpy as an integral part of Spyder...

**Comment by hoechenberger** [comment]
@cbrnr afaik their resources are super limited and they didn't have the time to test with qt6 yet

**Comment by hoechenberger** [comment]
@cbrnr https://github.com/spyder-ide/spyder/issues/20201

**Comment by hoechenberger** [comment]
@larsoner We could consider dropping Spyder and guiding users to the Spyder standalone installers. We'd need to provide some setup script to deploy a proper configuration to make it easy to use MNE from the MNE installers though ...

**Comment by larsoner** [comment]
Hah, apparently we left automerge enabled! But that's okay -- I think for 1.8 we probably should drop Spyder from our installers and point people toward [their standalone installers](https://docs.spyder-ide.org/current/installation.html#standalone-installers). I don't think it'll be too bad to configure it to use our Python distribution/kernel because we install `spyder-kernels` here. But it's something we should do for 1.8 docs. I'll open an issue in MNE-Python and tag for 1.8 so we don't forget.

**Comment by hoechenberger** [comment]
thanks Eric! the inclusion of MNELAB is worth it!

speaking of which, we should probably add it to the installer menu 

it's something we need to take care of in the mne-feedstock repo

**Comment by larsoner** [comment]
Somewhere we have a note about moving the installer menu stuff to this repo, is that still possible? If feel free to give it a shot, or if you know if another project that does it in the constructor I can look at their code and adapt. 

I'll look at Spyder and Napari at least to see if they have something to work from

**Pull request merged**